core/src/main/scala/akka/persistence/r2dbc/session/scaladsl/R2dbcSession.scala

```diff
@@ -33,9 +35,17 @@ object R2dbcSession {
 
   def withSession[A](system: ActorSystem[_], connectionFactoryConfigPath: String)(
       fun: R2dbcSession => Future[A]): Future[A] = {
-    val connectionFactory =
-      ConnectionFactoryProvider(system).connectionFactoryFor(connectionFactoryConfigPath)
-    val r2dbcExecutor = new R2dbcExecutor(connectionFactory, log, logDbCallsDisabled)(system.executionContext, system)
+    val connectionFactoryProvider = ConnectionFactoryProvider(system)
+    val connectionFactory = connectionFactoryProvider.connectionFactoryFor(connectionFactoryConfigPath)
+    val closeCallsExceeding =
+      connectionFactoryProvider
+        .connectionFactorySettingsFor(connectionFactoryConfigPath)
+        .poolSettings
+        .closeCallsExceeding
```

Comment:
I was pondering where to place the closeCallsExceeding setting. It has to be in the connection-factory section in reference.conf because it's used from this `withSession`, which isn't aware of pluginId or such.

It's not specific to a dialect, so therefore I placed it in the common `default-connection-pool`. However, it's not really related to the pool. Maybe it's better to place it directly in `connection-factory` and `ConnectionFactorySettings`?

Reply:
I don't think pool is that wrong, isn't it about keeping the pool healthy?

Reply:
yeah, good enough, not exposed to users, the config property would be the same